Revised Jira integration page to be more task-based (#395)
OctopusDeploy/docs: docs/api-and-integration/metadata/jira.md

1. Set the Octopus Server URL in Octopus Deploy.

    Navigate to the **{{Configuration,Nodes}}** page and ensure you have set the Server URI field to your Octopus Server's base URL. i.e., https://my-company.octopus.app/ or https://my-company-internal-name/

1. Configure the Octopus Deploy plugin for Jira.

    Navigate to the **{{Configuration,Settings,Jira Issue Tracker}}** page in Octopus, copy the **Octopus InstallationID**, and add it to Jira App configuration.

1. Update your Octopus environment settings.

    Navigate to **{{Infrastructure,Environments}}** to map your Octopus environments to Jira environment types. This is required so Jira can understand Octopus environments and track issue progress.

    Note: Jira environment types are a fixed list that cannot be edited.

When configured this integration will provide Jira with updates about the progress of Jira issues (work items) through the pipeline.

## Octopus Deployment Task Log

> When the Jira issue tracker is enabled and configured, you will see blocks similar to the following appear in the log during your deployments. These show the state updates Octopus is sending through to Jira, and if you expand them the details include information about the Jira issues for traceability.

Reviewer: ```suggestion
When the Jira issue tracker is enabled and configured, the task log will show the state updates Octopus is sending through to Jira. If you expand them the details include information about the Jira issues for traceability.
```

description: Configure Jira Software issue tracking with Octopus.
---

**Octopus 2019.4** introduced support to integrate Octopus with Jira Software (Jira Cloud). The integration provides release and deployment details from Octopus directly in Jira issues, making it possible to see if the issue has been associated with any deployments. This builds upon the functionality to [track metadata and work item](/docs/api-and-integration/metadata/index.md) information through your CI/CD pipeline.

This integration uses the [Octopus Deploy plugin for Jira](https://marketplace.atlassian.com/apps/1220376/octopus-deploy-for-jira) and this page describes how to connect and configure the two applications.

![Jira Deployments](jira-deployment.png "width=500")

:::hint
The Octopus Deploy plugin for Jira is only compatible with Jira Software Cloud as Jira Server (on-premises) does not support the APIs required to enable this functionality. There is limited support available as noted in our [Jira Server section](#jira-server-on-prem-support).
:::

## Connecting Jira and Octopus Deploy

1. Install the Octopus Deploy plugin in Jira.

>     Add the `Octopus Deploy for Jira` app from the [Atlassian Marketplace](https://marketplace.atlassian.com/apps/1220376/octopus-deploy-for-jira) and then click the 'Get Started' button to configure it.

Reviewer: Do we need to provide orientation here? i.e., From within your Jira administration dashboard...

Author: Yes, that's probably a good idea. It's not hard but we should probably add more detail.

    - Release Note Prefix: Specifying a 'release note prefix' tells Octopus to search through your Jira issue description for that value. If it finds a match, then it will use the issue description as the release note; otherwise, it will use the issue title. See [Release Notes](/docs/api-and-integration/metadata/release-notes-templates.md) for more information.

1. Set the Octopus Server URL in Octopus Deploy.

    Navigate to the **{{Configuration,Nodes}}** page and ensure you have set the Server URI field to your Octopus Server's base URL. i.e., https://my-company.octopus.app/ or https://my-company-internal-name/

1. Configure the Octopus Deploy plugin for Jira.

    Navigate to the **{{Configuration,Settings,Jira Issue Tracker}}** page in Octopus, copy the **Octopus InstallationID**, and add it to Jira App configuration.

1. Update your Octopus environment settings.

    Navigate to **{{Infrastructure,Environments}}** to map your Octopus environments to Jira environment types. This is required so Jira can understand Octopus environments and track issue progress.

    Note: Jira environment types are a fixed list that cannot be edited.

> When configured this integration will provide Jira with updates about the progress of Jira issues (work items) through the pipeline.

Reviewer: For the sake of this audience, I don't know that we need to include the phrase (work items).

Author: Agreed.
